### ISSUE TITLE
CIS scan tool support for cis v 1.5

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#24987]
**What kind of request is this (question/bug/enhancement/feature request):** enhancement

CIS Scan tool must support scan for CIS benchmark v 1.5

**Comment by leodotcloud** [comment]
These are the current pending failing tests that need to be looked into. Some of them can use the fixes from rke-cis-1.4 directly, others are totally new.

  |   |   |   |   |  
-- | -- | -- | -- | -- | --
Open accordion | Fail | 1.1.13 | Ensure that the admin.conf file permissions are set to 644 or more restrictive (Scored)Remediation: Run the below command (based on the file location on your system) on the master node. For example, chmod 644 /etc/kubernetes/admin.conf | Skip |  
Open accordion | Fail | 1.1.14 | Ensure that the admin.conf file ownership is set to root:root (Scored)Remediation: Run the below command (based on the file location on your system) on the master node. For example, chown root:root /etc/kubernetes/admin.conf | Skip |  
Open accordion | Fail | 1.1.15 | Ensure that the scheduler.conf file permissions are set to 644 or more restrictive (Scored)Remediation: Run the below command (based on the file location on your system) on the master node. For example, chmod 644 /etc/kubernetes/scheduler.conf | Skip |  
Open accordion | Fail | 1.1.16 | Ensure that the scheduler.conf file ownership is set to root:root (Scored)Remediation: Run the below command (based on the file location on your system) on the master node. For example, chown root:root /etc/kubernetes/scheduler.conf | Skip |  
Open accordion | Fail | 1.1.17 | Ensure that the controller-manager.conf file permissions are set to 644 or more restrictive (Scored)Remediation: Run the below command (based on the file location on your system) on the master node. For example, chmod 644 /etc/kubernetes/controller-manager.conf | Skip |  
Open accordion | Fail | 1.1.18 | Ensure that the controller-manager.conf file ownership is set to root:root (Scored)Remediation: Run the below command (based on the file location on your system) on the master node. For example, chown root:root /etc/kubernetes/controller-manager.conf | Skip |  
Open accordion | Fail | 1.2.34 | Ensure that encryption providers are appropriately configured (Scored)Remediation: Follow the Kubernetes documentation and configure a EncryptionConfig file. In this file, choose aescbc, kms or secretbox as the encryption provider. | Skip |  
Open accordion | Fail | 3.2.1 | Ensure that a minimal audit policy is created (Scored)Remediation: Create an audit policy file for your cluster. | Skip |  
Open accordion | Fail | 4.1.1 | Ensure that the kubelet service file permissions are set to 644 or more restrictive (Scored)Remediation: Run the below command (based on the file location on your system) on the each worker node. For example, chmod 644 $kubeletsvc | Skip |  
Open accordion | Fail | 4.1.2 | Ensure that the kubelet service file ownership is set to root:root (Scored)Remediation: Run the below command (based on the file location on your system) on the each worker node. For example, chown root:root $kubeletsvc | Skip |  
Open accordion | Fail | 4.1.3 | Ensure that the proxy kubeconfig file permissions are set to 644 or more restrictive (Scored)Remediation: Run the below command (based on the file location on your system) on the each worker node. For example, chmod 644 $proykubeconfig | Skip |  
Open accordion | Fail | 4.1.4 | Ensure that the proxy kubeconfig file ownership is set to root:root (Scored)Remediation: Run the below command (based on the file location on your system) on the each worker node. For example, chown root:root $proxykubeconfig | Skip |  
Open accordion | Fail | 4.1.7 | Ensure that the certificate authorities file permissions are set to 644 or more restrictive (Scored)Remediation: Run the following command to modify the file permissions of the --client-ca-file chmod 644 <filename> | Skip |  
Open accordion | Fail | 4.1.8 | Ensure that the client certificate authorities file ownership is set to root:root (Scored)Remediation: Run the following command to modify the ownership of the --client-ca-file. chown root:root <filename> | Skip |  
Open accordion | Fail | 4.1.9 | Ensure that the kubelet configuration file has permissions set to 644 or more restrictive (Scored)Remediation: Run the following command (using the config file location identied in the Audit step) chmod 644 $kubeletconf | Skip |  
Open accordion | Fail | 4.1.10 | Ensure that the kubelet configuration file ownership is set to root:root (Scored)Remediation: Run the following command (using the config file location identied in the Audit step) chown root:root $kubeletconf | Skip |  
Open accordion | Fail | 4.2.12 | Ensure that the RotateKubeletServerCertificate argument is set to true (Scored)Remediation: Edit the kubelet service file $kubeletsvc on each worker node and set the below parameter in KUBELET_CERTIFICATE_ARGS variable. --feature-gates=RotateKubeletServerCertificate=true Based on your system, restart the kubelet service. For example: systemctl daemon-reload systemctl restart kubelet.service | Skip |  
Open accordion | Fail | 5.1.5 | Ensure that default service accounts are not actively used. (Scored)Remediation: Create explicit service accounts wherever a Kubernetes workload requires specific access to the Kubernetes API server. Modify the configuration of each default service account to include this value automountServiceAccountToken: false | Skip |  
Open accordion | Fail | 5.3.2 | Ensure that all Namespaces have Network Policies defined (Scored)Remediation: Follow the documentation and create NetworkPolicy objects as you need them. | Skip |  
Open accordion | Fail | 5.6.4 | The default namespace should not be used (Scored)Remediation: Ensure that namespaces are created to allow for appropriate segregation of Kubernetes resources and that all new resources are created in a specific namespace. | Skip

**Comment by prachidamle** [comment]
On a cluster run CIS 1.5 Hardened profile run(hardened cluster)
Only Failures seen:
5.1.5 - Ensure that default service accounts are not actively used. (Scored)
5.6.4 - The default namespace should not be used (Scored) - currently on master-head Admin is not able delete default namespace
error seen:
5.3.2 - Ensure that all Namespaces have Network Policies defined (Scored)

Fixes for 5.1.5 and 5.6.4 fixes have been added; For 5.3.2 steps from hardening guide should be followed.

**Comment by prachidamle** [comment]
The API supports running CIS v1.4

**Comment by izaac** [comment]
I tested this with CIS-1.5 using the API in v2.4.4 and it seems that the Scan `5.3.2` will always fail as it hasn't the audit script yet.

```
Ensure that all Namespaces have Network Policies defined (Scored)Remediation: Follow the documentation and create NetworkPolicy objects as you need them.
```

https://github.com/rancher/security-scan/blob/master/package/cfg/rke-cis-1.5/policies.yaml#L95

**Comment by sangeethah** [comment]
Related UI issues:
https://github.com/rancher/rancher/issues/27867
https://github.com/rancher/rancher/issues/27866

**Comment by sowmyav27** [comment]
**Verified on 2.4-head - commit id: `b7ab53e95` and on master-head - commit id: `80bd60a89`**

- Default CIS scan version run on a k8s 1.18 cluster is CIS 1.5 when done through View In API --> runSecurityScan
- On the UI, default scan version available is RKE CIS 1.5 for a k8s 1.18 cluster
